core/client/db/src/light.rs

```diff
@@ -370,7 +370,7 @@ impl<Block> LightBlockchainStorage<Block> for LightStorage<Block>
 	fn import_header(
 		&self,
 		header: Block::Header,
-		authorities: Option<Vec<AuthorityIdFor<Block>>>,
+		cache_at: HashMap<Vec<u8>, Vec<u8>>,
```

Comment:
As a possible optimization (probably for future PRs): use `[u8; 4]` or `&'static [u8; 4]` as keys in this `HashMap` to avoid extra allocations. Pretty sure that the all keys in this map will be defined as static constants.

Reply:
https://github.com/paritytech/substrate/issues/2148